tuna/machine.py

```diff
@@ -283,31 +306,48 @@ def get_properties(self):
 
     return self.cpus, self.gpus
 
-  def write_file(self, contents, filename=None, is_temp=False):
+  def write_file(self, contents: bytes, filename: Union[bytes, Text] = None , \
+  is_temp: bool = False) -> Union[bytes, Text]:
     """
     Write a file to this machine containing contents
     """
+    cnx: Connection
+    ftp: Optional[paramiko.sftp_client.SFTPClient] = None
+    fout: SFTPFile
+    fout_2: BufferedWriter
+
+    t_filename: Union[str, 'os.PathLike[Any]']
     if is_temp:
       assert filename is None
-      _, filename = tempfile.mkstemp()
-    else:
-      assert filename is not None
+      _, t_filename = tempfile.mkstemp()
+      return t_filename
```

Comment:
if this returns here, the contents never gets written to the json, hence and empty json file and FIN will complain the json is not as expected. Please address by removing the return statement here